src/components/Auth/Auth.tsx

```diff
@@ -1,44 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
 import { Navigate, Outlet, useLocation } from 'react-router-dom';
-import api from '../../api/api';
-
-const LOCALSTORAGE_USER_NAME = 'zaaktypenbeheer_user';
+import { api } from '../../api/api';
 
 export interface AuthContextType {
   user: any;
   signIn: (formData: FormData, callback: VoidFunction) => void;
   signOut: (callback: VoidFunction) => void;
 }
 
-let AuthContext = React.createContext<AuthContextType>(null!);
+const AuthContext = React.createContext<AuthContextType>(null!);
 
 export function AuthProvider({ children }: { children: React.ReactNode }) {
-  let [user, setUser] = React.useState<any>(localStorage.getItem(LOCALSTORAGE_USER_NAME));
+  const [authState, setAuthState] = useState<any>({
+    user: null,
+    authChecked: false,
+  });
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      const user = await api.getUser();
+      setAuthState({
+        user,
+        authChecked: true,
+      });
+    };
+    if (!authState.authChecked) fetchUser();
+  }, []);
 
-  let signIn = async (formData: FormData, callback: VoidFunction) => {
-    await api.signIn(formData);
-    localStorage.setItem(LOCALSTORAGE_USER_NAME, formData.get('username') as string);
-    setUser(formData.get('username'));
+  const signIn = useCallback(async (formData: FormData, callback: VoidFunction) => {
+    const data = {
+      username: formData.get('username'),
+      password: formData.get('password'),
+    };
+    const user = await api.signIn(data);
+    setAuthState({
+      user,
+      authChecked: true,
+    });
     callback();
-  };
+  }, []);
 
-  let signOut = (callback: VoidFunction) => {
-    setUser(null);
+  const signOut = useCallback(async (callback: VoidFunction) => {
```

Comment:
Maybe these callbacks can be renamed to `onSignIn` and `onSignOut` to make it extra clear that they are callbacks?